compiler/rustc_smir/src/rustc_smir/mod.rs

```diff
@@ -287,6 +287,37 @@ fn rustc_generator_to_generator(
     }
 }
 
+fn rustc_inline_asm_operand_to_inline_asm_operand(
+    operand: &rustc_middle::mir::InlineAsmOperand<'_>,
+) -> stable_mir::mir::InlineAsmOperand {
+    use rustc_middle::mir::InlineAsmOperand;
+
+    match operand {
+        InlineAsmOperand::In { value, .. } => stable_mir::mir::InlineAsmOperand {
+            place: None,
+            operand: Some(rustc_op_to_op(value)),
+            raw_rpr: format!("{:?}", operand),
+        },
+        InlineAsmOperand::Out { place, .. } => stable_mir::mir::InlineAsmOperand {
+            place: place.map(|place| rustc_place_to_place(&place)),
+            operand: None,
+            raw_rpr: format!("{:?}", operand),
+        },
+        InlineAsmOperand::InOut { in_value, out_place, .. } => stable_mir::mir::InlineAsmOperand {
+            place: out_place.map(|place| rustc_place_to_place(&place)),
+            operand: Some(rustc_op_to_op(in_value)),
+            raw_rpr: format!("{:?}", operand),
+        },
+        InlineAsmOperand::Const { .. }
+        | InlineAsmOperand::SymFn { .. }
+        | InlineAsmOperand::SymStatic { .. } => stable_mir::mir::InlineAsmOperand {
+            place: None,
+            operand: None,
+            raw_rpr: format!("{:?}", operand),
```

Comment:
Only format this once. You can return the options from the match and only create the struct once